src/libstore/globals.hh

```diff
@@ -523,9 +523,11 @@ public:
         {"cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY="},
         "trusted-public-keys",
         R"(
-          A whitespace-separated list of public keys. When paths are copied
-          from another Nix store (such as a binary cache), they must be
-          signed with one of these keys. For example:
+          A whitespace-separated list of public keys. When paths are
+          copied from another Nix store (such as a binary cache), and
+          are not part of a fixed-output derivation (such as a
+          fetcher), they must be signed with one of these keys. For
+          example:
```

Comment:
```suggestion
          copied from another Nix store (such as a binary cache), they must be
          signed with one of these keys, unless they are content-addressed (like
          the outputs of fixed-output derivations). For example:
```

Nix doesn't really care about how the path was built, it just has to be content-addressed.
For instance you can `nix copy --from https://cache.ngi0.nixos.org/ /nix/store/yvk5yl9fid0zlxqk1xvvzn787d8gbh00-emacs-27.2` and it will work just fine, although the path doesn't come from a FO derivation